sdk/data/azcosmos/cosmos_location_cache.go

```diff
@@ -0,0 +1,298 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package azcosmos
+
+import (
+	"net/url"
+	"sync"
+	"time"
+)
+
+const defaultExpirationTime time.Duration = time.Minute * 5
+
+const (
+	none opType = iota
+	read
+	write
+	all
```

Comment:
Is `all` used?

Reply:
Yes. Currently, I used bitwise operations to treat this enum/iota as flags. So all = read&write